src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -555,13 +560,37 @@ private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SA
             log.warn("Record mapped off end of reference; making unmapped: " + aligned);
             SAMUtils.makeReadUnmapped(unaligned);
         } else if (isContaminant) {
-            log.warn("Record looks like foreign contamination; making unmapped: " + aligned);
-            // NB: for reads that look like contamination, just set unmapped flag and zero MQ but keep other flags as-is.
-            // this maintains the sort order so that downstream analyses can use them for calculating evidence
-            // of contamination vs other causes (e.g. structural variants)
+            if (numForeignContaminantWarnings < MAX_FOREIGN_CONTAM_WARN) {
+                log.warn("Record looks like foreign contamination ; making unmapped: " + aligned);
+                numForeignContaminantWarnings++;
+            }
+
+            if (numForeignContaminantWarnings == MAX_FOREIGN_CONTAM_WARN) {
+                log.warn("No more warnings regarding foreign contamination; limit reached.");
+            }
+
+            // For reads that look like contamination, set unmap. And reset the mapping information as
+            // keeping it violates the sam-spec and is not supported by some formats (CRAM)
```

Comment:
I talked to the folks who develop CRAM and they seem to want a change in SAM spec for that...but I'm a reasonable person...how about I (optionally) copy the mapping information to a tag and not remove it from the record itself?